From: Debian Games Team <pkg-games-devel@lists.alioth.debian.org>
Date: Sun, 4 Jan 2015 20:49:22 +0100
Subject: dynamic-link

Changes the static linking of external libraries to dynamic link
Original patch by Matthew Bekkema <mat8913ftw@gmail.com>
---
 cmake/Modules/FindJson.cmake | 24 +++++++++++-------------
 src/CMakeLists.txt           |  3 ---
 2 files changed, 11 insertions(+), 16 deletions(-)

diff --git a/cmake/Modules/FindJson.cmake b/cmake/Modules/FindJson.cmake
index a9178a2..1c0fe50 100644
--- a/cmake/Modules/FindJson.cmake
+++ b/cmake/Modules/FindJson.cmake
@@ -1,18 +1,16 @@
 # Look for json, use our own if not found
 
-#FIND_PATH(JSON_INCLUDE_DIR json.h)
+FIND_PATH(JSON_INCLUDE_DIR json/json.h /usr/include/jsoncpp)
 
-#FIND_LIBRARY(JSON_LIBRARY NAMES jsoncpp)
+FIND_LIBRARY(JSON_LIBRARY NAMES jsoncpp)
 
-#IF(JSON_LIBRARY AND JSON_INCLUDE_DIR)
-#	SET( JSON_FOUND TRUE )
-#ENDIF(JSON_LIBRARY AND JSON_INCLUDE_DIR)
+IF(JSON_LIBRARY AND JSON_INCLUDE_DIR)
+	SET( JSON_FOUND TRUE )
+ENDIF(JSON_LIBRARY AND JSON_INCLUDE_DIR)
 
-#IF(JSON_FOUND)
-#	MESSAGE(STATUS "Found system jsoncpp header file in ${JSON_INCLUDE_DIR}")
-#	MESSAGE(STATUS "Found system jsoncpp library ${JSON_LIBRARY}")
-#ELSE(JSON_FOUND)
-	SET(JSON_INCLUDE_DIR ${PROJECT_SOURCE_DIR}/json)
-	SET(JSON_LIBRARY jsoncpp)
-	MESSAGE(STATUS "Using project jsoncpp library")
-#ENDIF(JSON_FOUND)
+IF(JSON_FOUND)
+	MESSAGE(STATUS "Found system jsoncpp header file in ${JSON_INCLUDE_DIR}")
+	MESSAGE(STATUS "Found system jsoncpp library ${JSON_LIBRARY}")
+ELSE(JSON_FOUND)
+	MESSAGE(FAIL "Cannot find system-wide jsoncpp library")
+ENDIF(JSON_FOUND)
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 63b73a6..b308ad2 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -264,7 +264,6 @@ if(NOT USE_LUAJIT)
 	message (STATUS "LuaJIT not found, using bundled Lua.")
 	set(LUA_INCLUDE_DIR "${PROJECT_SOURCE_DIR}/lua/src")
 	set(LUA_LIBRARY "lua")
-	add_subdirectory(lua)
 endif(NOT USE_LUAJIT)
 
 mark_as_advanced(LUA_LIBRARY)
@@ -770,7 +769,6 @@ endif(USE_GETTEXT)
 
 if (SQLITE3_FOUND)
 else (SQLITE3_FOUND)
-	add_subdirectory(sqlite)
 endif (SQLITE3_FOUND)
 
 if (BUILD_CLIENT AND USE_FREETYPE)
@@ -779,7 +777,6 @@ endif (BUILD_CLIENT AND USE_FREETYPE)
 
 if (JSON_FOUND)
 else (JSON_FOUND)
-	add_subdirectory(json)
 endif (JSON_FOUND)
 
 #end
